tests/test_aws_generator.py

```diff
@@ -180,85 +231,96 @@ def setUp(self):
         self.assertIn(key, generator.AWS_COLUMNS)
         self.assertNotIn("key-that-has-not-been-added", generator.AWS_COLUMNS)
+        os.remove(test_config)
 
     def test_unknown_location(self):
         """Test that an unknown location doesn't result in stack trace."""
-        self.attributes["region"] = "Bad result"
+        test_args = self.test_config_kwargs
+        test_args["region"] = "Bad Result"
         key = "resourceTags/user:new-key"
-        tag_cols = {key}
-        two_hours_ago = (self.now - self.one_hour) - self.one_hour
-        generator = TestGenerator(two_hours_ago, self.now, self.payer_account, self.usage_accounts, tag_cols=tag_cols)
+        test_args["tags"]["key"] = key
+        test_config = create_test_config(**test_args)
+        generator = DataTransferGenerator(self.two_hours_ago, self.now)
         self.assertIn(key, generator.AWS_COLUMNS)
         self.assertNotIn("key-that-has-not-been-added", generator.AWS_COLUMNS)
+        os.remove(test_config)
 
 
 class TestRDSGenerator(AWSGeneratorTestCase):
     """Tests for the RDS Generator type."""
 
     def test_init_no_attributes(self):
         """Test the init wihout attributes."""
-        generator = RDSGenerator(
-            self.two_hours_ago, self.now, self.payer_account, self.usage_accounts, attributes={"empty": "dictionary"}
-        )
-        self.assertIsNotNone(generator._product_sku)
-        self.assertIsNotNone(generator._resource_id)
-        self.assertIsNotNone(generator._instance_type)
+        generator = RDSGenerator(self.two_hours_ago, self.now)
+        self.assertIsNotNone(generator.config[0].get("product_sku"))
+        self.assertIsNotNone(generator.config[0].get("resource_id"))
+        self.assertIsNotNone(generator.config[0].get("instance_type"))
 
     def test_init_with_attributes(self):
         """Test the unique init options for RDS."""
-        generator = RDSGenerator(
-            self.two_hours_ago, self.now, self.payer_account, self.usage_accounts, self.attributes
-        )
-        self.assertEqual(generator._product_sku, self.product_sku)
-        self.assertEqual(generator._resource_id, "i-" + self.resource_id)
-        self.assertEqual(generator._tags, self.tags)
-        self.assertEqual(generator._instance_type[:-1], tuple(self.instance_type.values()))
+        test_args = self.test_config_kwargs
+        test_args["generator"] = "RDSGenerator"
+        test_config = create_test_config(**test_args)
+        generator = RDSGenerator(self.two_hours_ago, self.now, user_config=test_config)
+        self.assertEqual(generator.config[0].get("product_sku"), self.product_sku)
+        self.assertIn(self.tags["key"], generator.config[0].get("tags").keys())
+        self.assertEqual(generator.config[0].get("resource_id"), self.resource_id)
+        self.assertEqual(tuple(generator.config[0].get("instance_type").values()), tuple(self.instance_type.values()))
```

Comment:
Do we need an `os.remove(test_config)` here?

Reply:
If so we are gonna need it these too: 
- `test_update_data`
- `test_init_with_attributes`

Reply:
Good catch. I'll fix those.